### PR TITLE
[PLAY-1038] Beta React-Router Scaffold/Setup

### PR DESCRIPTION
**Pull request by jasperfurniss**
**What does this PR do?** A clear and concise description with your runway ticket url.
https://nitro.powerhrg.com/runway/backlog_items/PLAY-1038

**Screenshots:** Screenshots to visualize your addition/change
Screenshot 2023-11-03 at 12 08 04 PM

**How to test?** Steps to confirm the desired behavior:
1. Go to 'beta/kits'
2. See view render in react with react-router.
3. Click the word Avatar and see the Component Show Page.

#### Checklist:
- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **DEPLOY** I have added the `milano` label to show I'm ready for a review.
- [x] **TESTS** I have added test coverage to my code.

### CODE REVIEW

File: playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const GuidesNavItems = [
```
Comment (jasoncypret) [inline playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts:1]: I've very concerned this is not coming from the back end. Lots of work was put into having this dynamically available. This is available from `DOCS` in the initializer

File: playbook-website/app/javascript/components/LayoutRight/index.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  Layout,
+  FlexItem,
+  Flex,
+  Background,
+  Button,
+  Title,
+  Body,
+} from "playbook-ui"
+
+export default function LayoutRight({ isMobile, dark }) {
```
Comment (jasoncypret) [inline playbook-website/app/javascript/components/LayoutRight/index.tsx:17]: Should this be specific to the content/page? LayoutRight sounds like a generic component that will load more of the site in. When this as is won't be loaded when we move on to "kit show" for example.  Should this be "kit index"?

File: playbook-website/app/javascript/components/App.tsx [hunk 3]
```diff
@@ -1,17 +1,29 @@
 import React, { useState, useEffect } from "react"
-import { BrowserRouter as Router, Route, Switch } from "react-router-dom"
-import MainSidebar from "./MainSidebar"
-import { Layout, Background, Icon, Title, Body, Button, Flex, FlexItem } from "playbook-ui"
-import HeaderMobile from "../images/pb_generic_header.jpg"
-import HeaderDesktop from "../images/pb_generic_header_desktop.jpg"
-import Footer from "./Footer"
-import DarkModeToggle from "./DarkModeToggle"
+import { Outlet } from "react-router-dom"
+import ReactSidebar from "./ReactSidebar"
```
Comment (jasoncypret) [inline playbook-website/app/javascript/components/App.tsx:3]: React seems to be unimportant here. Can we drop/change that? I'd be more curious to define this as the site navigation or something more descriptive.

**Review by jasoncypret — COMMENTED** [review]
Good start. Early feedback is that some of the naming should be reconsidered.

Thoughts/Observations:
**Global things** should be in a folder structure that reflects the global nature. The naming should be accurate to what function/purpose it served

**Page/View specific things** should be in a folder structure that reflects the Page/View they beling with. The naming should be accurate to what function/purpose it served. Components that belong to a page/view/state only should be named clearly and location should be near the parent.

Idea:
You might also consider adding a readme in all the main folders that describe what belongs in said folder and how the naming should be observed in the area. I worry if you are not as involved in the next stories the structure will devolve/deteriorate.

Question:

File: playbook-website/app/javascript/components/App.tsx (hunk 3 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/App.tsx:3]: Sure. I can change that.

File: playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts (hunk 1 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts:1]: Okay no problem. I'll render it from there.

File: playbook-website/app/javascript/components/LayoutRight/index.tsx (hunk 2 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/LayoutRight/index.tsx:17]: The kit show page will still load... inside of the LayoutRight view. 
It loads as an `<Outlet />` in react-router. 

I separated this out because my App.tsx was getting too large. 
Since the page always renders a LeftLayout, which includes the Sidebar that we have, I went ahead and called them LeftLayout and RightLayout.

File: playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts (hunk 1 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts:1]: The original reason we were not able to use this was that since this method pulls the info from the url, it was returning null when on on a guides page.....this meant that our sidebar would not populate the nav items under 'getting started' if we were not on a guides/ page.

We did not have this issue when it was built because we had a separate, custom sidebar on the guides pages that only rendered on those pages. 

@jasoncypret 

Would you like me to investigate changing/updating the DOCS area, so that it will work with our pages now in this story? Or save it for a diff story?

Comment (jasperfurniss) [reply playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts:1]: I copied this over from Nida's sidebar that is on prod. ^^ And that's some of the reasoning.

Comment (jasperfurniss) [reply playbook-website/app/javascript/components/ReactSidebar/MenuData/GuildesNavItems.ts]: I got it working now with the back end data.

File: playbook-website/app/javascript/components/PbKitFetch/index.tsx [hunk 4]
```diff
@@ -11,8 +11,6 @@ const PbKitFetch = ({ kit }) => {
     fetch('/kits/avatar/rails_raw')
       .then((response) => response.text())
       .then((data) => {
-        console.log(data)
-        debugger
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/PbKitFetch/index.tsx:15]: 😳

File: playbook-website/app/javascript/components/Website/index.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import Sidebar from "./src/layouts/Sidebar"
+import LayoutRight from "./src/layouts/LayoutRight"
+
+function App() {
+  const [kits, setKits] = useState([])
+  const [dark, setDark] = useState(false)
+  const [isMobile, setIsMobile] = useState(window.innerWidth < 768)
+
+  useEffect(() => {
+    fetch("/beta/kits.json")
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/index.tsx:12]: This `fetch` seems a bit coupled here with the view itself. My suggestion would be to abstract any `fetch` calls out to an API layer.

File: playbook-website/app/javascript/components/Website/index.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      .then((data) => {
+        setKits(data.kits)
+        setDark(data.dark)
+      })
+      .catch((error) => {
+        console.log(error)
+      })
+
+    window.addEventListener("resize", () => {
+      setIsMobile(window.innerWidth < 768)
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/index.tsx:23]:
```tsx
window.innerWidth < 768
```

This part really should be a constant so that it is DRYer

File: playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React from "react"
+import { useParams, useSearchParams, NavLink } from "react-router-dom"
+import {Title} from "playbook-ui"
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx:3]: This is linty iirc the inner curlys should have padding

File: playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  const formatName = (name: any) => {
+    return name
+      .split("_")
+      .map((word: any) => word[0].toUpperCase() + word.slice(1))
+      .join(" ")
+  }
+
+  return (
+    <div className='pl_md'>
+      <NavLink to={`/beta/kits/${name}`}>
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx:20]:
```tsx
      `/beta/kits/${name}`
```

The `beta/kits` part of this link is used in other places. It might make sense to abstract this away to an import so that you can change it in one place instead of multiple.

File: playbook-website/app/javascript/components/Website/src/hooks/RedirectToType.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  const [searchParams] = useSearchParams()
+  const queryType = searchParams.get("type")
+
+  //  Supports our old routing, keeping our routes backwards-compatible.
+  if (queryType) {
+    return <Navigate to={`/beta/kit_category/${name}/${queryType}`} replace />
+  }
+
+  // If there's no queryType, navigate to the defaultType (react).
+  return <Navigate to={`/beta/kit_category/${name}/react`} replace />
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/hooks/RedirectToType.tsx:15]: I think that, instead, you should dynamically build the path and only return one instance of `<Navigate...`

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/MenuData/GuidelinesNavItems.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
+export const VisualGuidelinesItems = [
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/layouts/Sidebar/MenuData/GuidelinesNavItems.ts:1]: Pascal case here makes the constant appear like a class instead which can be confusing. It might for easier reading if `VISUAL_GUIDELINES_ITEMS=` was used instead. This is a predictable way to declare constants and is clear to other devs.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx [hunk 11]
```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect, useContext } from "react"
+import { linkFormat } from "../../../../../../utilities/website_sidebar_helper"
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx:2]: This path is a little rough and I wonder if a Webpack alias is a better fit. Relative paths are hard to work with.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  //click on nested items
+  const handleSubItemClick = (subLinkIndex, sublink, Index) => {
+    updateTopLevelNav(parentIndex)
+    updateKitsNav(Index)
+    collapsibles[Index][2](false) // Use the setCollapsed function
+  }
+
+  if (typeof link === "object") {
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx:41]: This is one of those areas that Sandy showed us how dependency injection can help clarify the contract of a class (React component in this case) so that the results are more predictable. If we have to check for type with `typeof` that is fine, but consider doing that beforehand and perhaps passing in the `RoutedNavItem` as the dependency.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+  let menuItems: { [key: string]: string }[] | string[] = [];
+
+  //conditionally render navitems depending on name
+  if (name === "Tokens & Guidelines") {
+    menuItems = VisualGuidelinesItems;
+  } else if (name === "UI Samples" && samples) {
+    menuItems = samplesMenu;
+  } else if (name === "Getting Started") {
+    menuItems = GuidesNavItems;
+  }
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx:44]: Consider using an object map here instead of conditionals:

```tsx
objMap = {
  "Tokens & Guidelines": VisualGuidelinesItems,
  "UI Samples": samplesMenu,
  "Getting Started": GuidesNavItems
}

menuItems = objMap[name]
```

Doing this would also potentiate the ability to make use of dependency injection.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/TopLevelNavItems.tsx [hunk 14]
```diff
@@ -0,0 +1,146 @@
+import React, { useEffect } from "react"
+import { NavItem, useCollapsible } from "playbook-ui"
+import { KitsNavItem} from "./NavComponents/KitsNavComponent"
+import { SideBarNavItems } from "./MenuData/SidebarNavItems"
+import { OtherNavItems } from "./NavComponents/OtherNavComponent"
+import RoutedNavItem from "./RoutedNavItem"
+import { useLocation, useParams, matchRoutes } from "react-router-dom"
+
+// const currentURL = window.location.pathname + window.location.search
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/layouts/Sidebar/TopLevelNavItems.tsx:9]: Remove

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/index.tsx [hunk 15]
```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react"
+import { Nav, useCollapsible, Image, Badge, Flex } from "playbook-ui"
+import { TopLevelNavItem } from "./TopLevelNavItems"
+// @ts-ignore
```
Comment (thestephenmarshall) [inline playbook-website/app/javascript/components/Website/src/layouts/Sidebar/index.tsx:4]: Why is this needed?

**Review by thestephenmarshall — COMMENTED** [review]
This is a lot of good work here. I just left some suggestions that might help tighten things up for future use. Good job so far!

File: playbook-website/app/javascript/components/Website/README.md [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+
+See the READMEs of all the sub-folders:
+
+- [assets](./src/assets/README.md)
+- [components](./src/components/README.md)
+- [hooks](./src/hooks/README.md)
+- [layouts](./src/layouts/README.md)
+- [pages](./src/pages/README.md)
+
+The routes are located at: `packs/app.js`.
```
Comment (nidaqg) [inline playbook-website/app/javascript/components/Website/README.md:13]: love the documentation! This will make working in these files much easier for other devs on future stories!

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx (hunk 13 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx:44]: 100%. Good call.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/index.tsx (hunk 15 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/layouts/Sidebar/index.tsx:4]: It might not be anymore, since I added a custom tsconfig.json for the website. I'll try removing! Thanks dude!

File: playbook-website/app/javascript/components/Website/index.tsx (hunk 6 above)
Comment (jasoncypret) [reply playbook-website/app/javascript/components/Website/index.tsx:23]: Can we import them? I see [we have an export](https://github.com/powerhome/playbook/blob/master/playbook/app/pb_kits/playbook/tokens/exports/_screen_sizes.scss)

`$screen-md-min`

File: playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx (hunk 8 above)
Comment (jasoncypret) [reply playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx:20]: Worth considering. We will swap this URL with the live one at some point.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx [hunk 17]
```diff
@@ -0,0 +1,75 @@
+import React from "react";
```
Comment (jasoncypret) [inline playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx:1]: What is an OtherNavComponent? Can we describe this file better?

File: playbook-website/app/javascript/components/Website/src/hooks/RedirectToType.tsx (hunk 9 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/hooks/RedirectToType.tsx:15]: Cool. I like it. I'll do that. This scaffold is really a future view anyways, and isn't being utilized yet for my current story--- but i'll go ahead and make that change as well.

File: playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx [hunk 18]
```diff
@@ -0,0 +1,25 @@
+import React from "react"
```
Comment (jasoncypret) [inline playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx:1]: How is this scaffold used? I'm not sure I'm following your naming. Seems like this is only used on the components pages/views?

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx (hunk 11 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx:2]: Made this much better. I had copied many of these files over in prep for upcoming work. 
Went ahead and created a hook, and co-located it much closer to this file, so the import doesn't look too too crazy now.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx (hunk 12 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/KitsNavComponent.tsx:41]: Gonna leave this for now, and wrestle with it in an upcoming story. Currently, none of the files inside of the Sidebar folder are being used in the beta view yet. 
I'll keep this note though for when we do try to implement this work.

File: playbook-website/app/javascript/components/Website/src/pages/README.md [hunk 19]
```diff
@@ -0,0 +1,5 @@
+# Pages Folder
```
Comment (jasoncypret) [inline playbook-website/app/javascript/components/Website/src/pages/README.md:1]: This is great. But knowing Show is going to get complex...do you want to provide a folder here showing how to group by page?

File: playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx (hunk 18 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/components/Scaffold.tsx:1]: It's not currently being used. I'll remove. 
I have it referenced for when we do the categories view in a follow-up story. But for now, since we aren't using, I'll go ahead and remove this file.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/MenuData/GuidesNavItems.ts [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  {
+    name: "React Setup",
+    link: "/guides/getting_started/react_setup",
+  },
+  {
+    name: "Ruby on Rails Setup",
+    link: "/guides/getting_started/ruby_on_rails_setup",
+  },
+];
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/layouts/Sidebar/MenuData/GuidesNavItems.ts:18]: I attempted to use the data that you had created. However, the controller that this data is in, the data only becomes available on visiting of the actual url (since it's a different one that my current view). In other words, my JSON was not being populated with the data, until i actual went to those pages, and then returned back to my beta/kits.json (then, i would see the data).

My plan was to separate the data into a rails concern that could be shareable across multiple controllers. And then bring that item into my controller as an include at the top. However, I was going to do this in a follow-up story because it ended up taking a good bit of time.

File: playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx (hunk 17 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/layouts/Sidebar/NavComponents/OtherNavComponent.tsx:1]: This isn't being used, and was copied over from Nida's work. I'll go ahead and remove this until we implement the react sidebar. For now, per your request, we are referencing the rails version that is rendered via her `MainSidebar` folder still, so that all the links redirect users to the legacy views.

File: playbook-website/app/javascript/components/Website/src/pages/README.md (hunk 19 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/pages/README.md:1]: I had that in the beginning but put back because the scope shifted to be much smaller based on our conversation. 

If you have a mockup that is finished, I can definitely take a look, and spec out the structure a little more here. 

Is the latest somewhere here? https://www.figma.com/file/8GHYnjKWp7jbUmCWWa7Yvt/Playbook-Site-Vision-2023?type=design&node-id=711-19827&mode=design&t=U25L77kO2e67v0ln-0

File: playbook-website/app/javascript/components/Website/src/components/README.md [hunk 21]
```diff
@@ -0,0 +1,5 @@
+# Components Folder
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/components/README.md:1]: My thought was that all re-usable components should live in the components folder, and they will be used in `pages/` files. So, a pages file might be ... ComponentShow.tsx or KitShow.tsx... and will use predominately components from the shared `components` folder. 

HeroImage I'd say belongs inside of this folder. But DarkModeToggle does feel odd... because it'll probably only live in one specific `layouts/{FILE}`. I'll keep marinating on where to put that one.

File: playbook-website/app/controllers/pages_controller.rb [hunk 22]
```diff
@@ -13,12 +13,13 @@ class PagesController < ApplicationController
   before_action :set_show_sidebar, only: %i[kits kit_category_show_rails kit_category_show_react kit_show_react kit_show_rails rails_in_react kit_show_demo kit_show_new visual_guidelines kit_show_swift home]
 
   def react_app
-    @kit = params[:name]
     @kits = MENU["kits"]
     @dark = cookies[:dark_mode] == "true"
+    @type = params[:type]
+
     respond_to do |format|
       format.html { render template: "react_app", layout: false }
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/controllers/pages_controller.rb:21]: Taking a look at this part now.

File: playbook-website/app/views/react_app.html.erb [hunk 23]
```diff
@@ -8,9 +8,19 @@
     <div class="pb--page--dark-mode-toggle-mobile">
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/views/react_app.html.erb:8]: I think the version you were code-reviewing, for some reason, was outdated. It contains everything that the application.html.erb had except the sidebar. But i'll take a look at if I can indeed use a layout for this instead.

**Review by thestephenmarshall — APPROVED** [review]

**Review by nidaqg — DISMISSED** [review]

File: playbook-website/app/views/layouts/application_beta.html.erb [hunk 24]
```diff
@@ -4,11 +4,20 @@
     <%= render 'layouts/head'%>
   </head>
   <body class="<%= @body_class %> <%= controller_name %> <%= action_name%>">
-    <%= render 'layouts/mobile_nav'%>
+
```
Comment (jasoncypret) [inline playbook-website/app/views/layouts/application_beta.html.erb:7]: Unsure if this is related but this is not desirable  https://share.zight.com/JrummNbA

File: playbook-website/app/javascript/components/Website/src/hooks/useFormatter.ts [hunk 25]
```diff
@@ -0,0 +1,10 @@
+export const useFormatter = (item) => {
+    const linkTitle = Array.isArray(item) ? item[0] : item
+    const replaceUnderscore = linkTitle
+      .replace(/_/g, ' ')
+      .split(' ')
+      .map(word => word.charAt(0).toUpperCase() + word.slice(1)) 
+      .join(' ')
+    return replaceUnderscore
+  }
+
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/hooks/useFormatter.ts:10]: Yes. I'll fix in a future story.

File: playbook-website/app/javascript/components/Website/index.tsx [hunk 26]
```diff
@@ -0,0 +1,15 @@
+import React from "react"
+import LayoutRight from "./src/layouts/LayoutRight"
+import { useLoaderData } from "react-router-dom"
+import { useMediaQuery } from "./src/hooks/useMediaQuery"
+
+function App() {
+  const { kits, dark } = useLoaderData()
+  const isMobile = useMediaQuery("(max-width: 768px)")
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/index.tsx:8]: I'll try.

File: playbook-website/app/javascript/components/Website/src/pages/ComponentList.tsx [hunk 27]
```diff
@@ -0,0 +1,19 @@
...
+
+  return (
+    <div className='pl_md'>
+      {!outlet && (
+        <NavLink to="avatar">
+          <Title paddingTop={isMobile ? "md" : ""} text={"Avatar"} size='2' />
+        </NavLink>
+      )}
+      <Outlet context={isMobile} /> 
+    </div>
```
Comment (jasoncypret) [inline playbook-website/app/javascript/components/Website/src/pages/ComponentList.tsx:17]: Can we clean this up in the next PR? I'd like to use a kit and a prop instead of direct classes

Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/pages/ComponentList.tsx:17]: Yes. I will do that in a follow-up story. 
This whole section will most likely be recoded, regardless. 
This is not final code, by any means.

File: playbook-website/app/javascript/components/Website/index.tsx (hunk 26 above)
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/index.tsx:8]: We don't export our tokens via JS in our dist. 
We have a hack for this that we use on our current site, and that we used in powerstory. 
Would you like me to use that hacky code? 

Or would you prefer we get a story on the board to provide JS exports for our tokens?

File: playbook-website/app/javascript/components/Website/src/pages/ComponentList.tsx (hunk 27 above)
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/src/pages/ComponentList.tsx]: Fixed this @jasoncypret ! Used `<Flex>` kit instead!

File: playbook-website/app/javascript/components/Website/index.tsx (hunk 26 above)
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook-website/app/javascript/components/Website/index.tsx]: Fixed this @jasoncypret ! Used our global responsive props, and got it working! So no custom hook required!

**Review by nidaqg — APPROVED** [review]
